### PR TITLE
Add Python constructor and comparison operator for ErtTemplates

### LINKED ISSUE #595: Python constructor and comparison for templates

### PR DESCRIPTION
**Pull request by mortalisk**
Resolves #595

### CODE REVIEW

**Comment by mortalisk** [comment]
retest this please

**Comment by mortalisk** [comment]
retest this please

File: lib/include/ert/job_queue/ext_job.hpp [hunk 1]
```diff
@@ -92,7 +92,6 @@ void                    ext_job_add_environment(ext_job_type *ext_job , const ch
 void                    ext_job_clear_environment( ext_job_type * ext_job );
 hash_type             * ext_job_get_environment( ext_job_type * ext_job );
 int                     ext_job_set_private_args_from_string( ext_job_type * ext_job , const char * arg_string );
-const char            * ext_job_get_private_args_as_string( ext_job_type * ext_job );
 const char *            ext_job_get_license_path(const ext_job_type*);
 //const char            * ext_job_get_arglist_as_string( ext_job_type * ext_job );
```
Comment (ManInFez) [inline lib/include/ert/job_queue/ext_job.hpp:96]: remove commented code

File: python/res/enkf/ert_templates.py [hunk 2]
```diff
@@ -15,27 +15,48 @@
...
 
-    def __init__(self, parent_subst, config_content):
+    def __init__(self, parent_subst, config_content=None, config_dict=None):
+        if not ((config_content is not None) ^ (config_dict is not None)):
+            raise ValueError('ErtTemplates must be instantiated with exactly one of config_content or config_dict')
 
-        c_ptr = self._alloc(parent_subst, config_content)
+        if config_dict is not None:
+            c_ptr = self._alloc_default(parent_subst)
+            super(ErtTemplates, self).__init__(c_ptr)
```
Comment (ManInFez) [inline python/res/enkf/ert_templates.py:42]: Should there be a test to see if c_ptr is None here as well?

File: python/tests/res/enkf/test_ert_templates.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        self.res_config = ResConfig(user_config_file=self.filename)
+
+    def tearDown(self):
+        del self.work_area
+
+    def test_all_templates_exist_with_correct_properties(self):
+        templates = ErtTemplates(self.res_config.subst_config.subst_list, config_dict=self.config_data)
+        template_names = templates.getTemplateNames()
+        configured_target_files = [t[0] for t in self.config_data[ConfigKeys.RUN_TEMPLATE]]
+        assert all(x in template_names for x in configured_target_files)
```
Comment (ManInFez) [inline python/tests/res/enkf/test_ert_templates.py]: could also use set(template_names) == set(configured_target_files)

File: python/tests/res/enkf/test_ert_templates.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        self.assertEqual(len(template_names), len(configured_target_files))
+        for configured_template in self.config_data[ConfigKeys.RUN_TEMPLATE]:
+            template = templates.get_template(configured_template[0])
+            self.assertEqual(template.get_template_file(), os.path.join(self.work_area.get_cwd(), configured_template[0]))
```
Comment (ManInFez) [inline python/tests/res/enkf/test_ert_templates.py:58]: maybe a __equal__ operator in templates could be used?

**Review by ManInFez — APPROVED** [review]
Some minor comments, but looks good overall.

File: python/tests/res/enkf/test_ert_templates.py (hunk 4 above)
Comment (mortalisk) [reply python/tests/res/enkf/test_ert_templates.py:58]: This is checking that the configuration results in correct template objects, so comparison isn't going to work I think. Or did I misunderstand you?

Comment (ManInFez) [reply python/tests/res/enkf/test_ert_templates.py:58]: I think I just misread the code, as I thought configured_template was a template_type object

**Pull request merged**